src/ir/item.rs

```diff
@@ -825,6 +826,50 @@ impl Item {
             _ => None,
         }
     }
+
+    /// Returns whether the item is a constified module enum
+    fn is_constified_enum_module(&self, ctx: &BindgenContext) -> bool {
```

Comment:
So what you're trying to do here is resolving through `ResolvedTypeRef`s, but not through `Alias`, right?

If so, instead of reinventing the wheel, you can do something like:

```rust
self.id.into_resolver()
    .through_type_refs()
    .resolve(ctx)
```

And keep the comment on why not through type alias.

We should arguably unify that and the `canonical_type` business, but that also handles some template stuff that may be nontrivial, so it's worth doing on a followup.